src/context.ts

```diff
@@ -107,61 +113,55 @@ export class Context
 
     public async getSelectedPackage(): Promise<packages.Package>
     {
-        let selectedRoot: string;
+        let selectedRoot: string | undefined;
         let folders = this._vscode.workspaceFolders;
 
         if (folders && folders.length > 0)
         {
             const selectionMode = this.packageSelectionMode;
-            if (selectionMode == "manual")
-            {
+            if (selectionMode == "manual") {
                 let root = this.rockSelectedPackage;
                 if (root)
                 {
                     if (exists(folders, root))
                         selectedRoot = root;
                 }
-            } else
-            {
+            }
+            else {
                 if (folders.length == 1 && folders[0].uri.scheme == 'file')
                     selectedRoot = folders[0].uri.fsPath;
 
-                const editor = this._vscode.activeTextEditor;
-                if (editor)
-                {
-                    const resource = editor.document.uri;
-                    if (resource.scheme === 'file')
-                    {
-                        const folder = this._vscode.getWorkspaceFolder(resource);
-                        if (folder)
-                            selectedRoot = folder.uri.fsPath;
-                    }
+                const currentDocumentURI = this._vscode.activeDocumentURI;
+                if (currentDocumentURI && currentDocumentURI.scheme === 'file') {
+                    const folder = this._vscode.getWorkspaceFolder(currentDocumentURI);
+                    if (folder)
+                        selectedRoot = folder.uri.fsPath;
                 }
             }
-            if (!selectedRoot && exists(folders, this._lastSelectedRoot))
+            if (!selectedRoot && this._lastSelectedRoot && exists(folders, this._lastSelectedRoot))
                 selectedRoot = this._lastSelectedRoot;
         }
         this._lastSelectedRoot = selectedRoot;
-        return this._packageFactory.createPackage(selectedRoot, this);
+        if (selectedRoot) {
```

Comment:
In principle, I don't have a problem with this but such logic belongs to the factory IMO (packages with a null/undefined path are invalid). It doesn't make much difference right now because `Context` is the only thing creating package instances but if/when we start using createPackage() in other places we will find ourselves repeating this check over and over again. Thoughts?

Reply:
I am trying to separate the code paths where a package is expected to be valid and places where it isn't. I think that we can reconsider *when* this check starts appearing a lot. Until then, I'd rather be stricter.

Reply:
Fair enough..